### ISSUE TITLE
[v3.15.0] App dies when hitting record

### ISSUE THREAD

**Issue by writingpoli** [OpenTracksApp/OpenTracks#613]
**Describe the bug**
App dies when trying to start tracking walk.

**To Reproduce**
1. Open app
2. Click on record
3. See error

# Workaround:
Go to settings and __disable "Latitude/Longitude"__.

**Technical information**
 - LG thinQ

(@dennisguse Removed unneeded technical info)

**Comment by dennisguse** [comment]
Which version? And can you provide a logcat?
The info above (sadly) does not help for finding the issue.

**Comment by oceyral** [comment]
Hi @dennisguse ,

Same thing here on app version 3.15.0, android 10 : app crash when hitting record. Empty track segments are created.

Not quite comfortable about sharing logcat, as I don't know what informations are in there, nor how to filter it, however looking at it briefly, the culprit seems obvious : 

```
FATAL EXCEPTION: main
Process: de.dennisguse.opentracks, PID: 24582
java.lang.NullPointerException: Attempt to invoke virtual method 'double java.lang.Double.doubleValue()' on a null object reference
	at de.dennisguse.opentracks.content.data.TrackPoint.getLatitude(TrackPoint.java:226)
	at de.dennisguse.opentracks.fragments.StatisticsRecordingFragment.setLocationValues(StatisticsRecordingFragment.java:446)
	at de.dennisguse.opentracks.fragments.StatisticsRecordingFragment.updateUI(StatisticsRecordingFragment.java:259)
	at de.dennisguse.opentracks.fragments.StatisticsRecordingFragment.lambda$onTrackUpdated$3$StatisticsRecordingFragment(StatisticsRecordingFragment.java:164)
	at de.dennisguse.opentracks.fragments.-$$Lambda$StatisticsRecordingFragment$qTCJqbKQ9ik7N-JSKTl9FPND1dg.run(Unknown Source:4)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7615)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:964)
```

This is quite bad, as the app can't even be rolled back to a previous version because of the database migration.

**Comment by dennisguse** [comment]
@oceyral Thanks!
That was sufficient!

The problem occurs with v3.15.0 if you Settings->Show Latitude/Longitude activated.
If you deactivate, the crash does not occur.

I will create a new release today!
The F-Droid release might take 2 weeks.

**Comment by oceyral** [comment]
Great ! Thank you for all the time and effort you dedicate to this project !

**Comment by dennisguse** [comment]
Issues is fixed and released in v3.15.1
F-Droid availability will take some days.

PS: there was actually another bug: lat/lon where actually never shown as the UI considered the data old.
This is fixed as well.

**Comment by dennisguse** [comment]
v3.15.1 is live on F-Droid since 03-Mar-2021.
